app/forms/piv_cac_proofing_form.rb

```diff
@@ -0,0 +1,50 @@
+class PivCacProofingForm
+  include ActiveModel::Model
+  include PivCacFormHelpers
+
+  attr_accessor :x509_dn_uuid, :x509_dn, :token, :error_type, :nonce, :user, :key_id, :first_name,
+                :last_name, :cn
+
+  validates :token, presence: true
+  validates :nonce, presence: true
+
+  def submit
+    success = valid? && valid_submission?
+
+    errors = error_type ? { type: error_type } : {}
```

Comment:
where does `error_type` get set? I'm not seeing it in this diff or in the included module

Reply:
PivCacFormHelpers